.github/workflows/CI.yaml

```diff
@@ -0,0 +1,258 @@
+
+        export counter=1
+        while [[ $(gcloud compute ssh --zone "us-central1-a" "test-gitian-$random"  --tunnel-through-iap --project "${{ secrets.GCP_PROJECT_ID_PROD }}" --command="ls -la" &>/dev/null || echo "re-try") == "re-try" && counter -lt 60 ]]
+        do
+          echo "attemp number: $counter"
+          export counter=$((counter+1))
+          if [ $counter -eq 60 ]; then gcloud compute instances delete "test-gitian-$random" --project "${{ secrets.GCP_PROJECT_ID_PROD }}" --zone "us-central1-a" --delete-disks=all; exit 1; fi
+          sleep 5
+        done
+
+        IFS='/' read -r -a array <<< "${{ github.event.label.name }}"
+
+        git clone -b ${array[2]} https://github.com/${array[0]}/${array[1]}.git
+        cd zcash/contrib/gitian-descriptors
+        wget -c https://github.com/mikefarah/yq/releases/download/v4.28.2/yq_linux_amd64
+        chmod +x yq_linux_amd64
+        export ZCASH_GITIAN_VERSION=$(cat gitian-linux-parallel.yml | ./yq_linux_amd64 .name)
+        cd ../../..
+
+
+        cat <<EOF > ./script.sh
+        apt update;
+        apt install ca-certificates curl gnupg lsb-release zsh software-properties-common wget git vagrant python3-venv direnv python3-pip linux-headers-\$(uname -r) ansible -y;
+        mkdir -m 0755 -p /etc/apt/keyrings;
+        curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes;
+        echo "deb [arch=\$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \$(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+        apt update;
+        apt install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin -y;
+        apt-add-repository "deb http://download.virtualbox.org/virtualbox/debian \$(lsb_release -sc) contrib";
+        wget -q https://www.virtualbox.org/download/oracle_vbox_2016.asc -O- | apt-key add -;
+        apt update
+        apt install virtualbox-6.1 -y;
+        eval "\$(direnv hook bash)";
+        cd source
+        cp .env.example .env
+        cp .envrc.example .envrc
+        /usr/bin/python3 -m venv ./local/python_venv;
+        echo "load_prefix local/python_venv" >> .envrc;
+        export VERSION="${array[2]}"
+        echo "ZCASH_VERSION=\$VERSION" >> .env;
+        echo "ZCASH_GIT_REPO_URL=https://github.com/${array[0]}/${array[1]}" >> .env;
+        cat .env
+        direnv allow;
+        pip3 install --upgrade pip;
```

Comment:
Hmm, I think this will download unpinned files. I know it's upgrading the pip in the virtualenv, but is that even necessary? The default one should work fine, shouldn't it?

Reply:
I think something was breaking if it wasn't updated (this code is legacy). I'll run the test without this line and see what happens